ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/AbstractCallOperation.java

```diff
@@ -146,55 +150,89 @@ protected AbstractCallOperation(
   protected abstract boolean isStatic(MessageFrame frame);
 
   @Override
-  public void execute(final MessageFrame frame) {
-    frame.clearReturnData();
-
-    final Address to = to(frame);
-    final Account contract = frame.getWorldState().get(to);
-
-    final Account account = frame.getWorldState().get(frame.getRecipientAddress());
-    final Wei balance = account.getBalance();
-    if (value(frame).compareTo(balance) > 0 || frame.getMessageStackDepth() >= 1024) {
-      frame.expandMemory(inputDataOffset(frame).intValue(), inputDataLength(frame).intValue());
-      frame.expandMemory(outputDataOffset(frame).intValue(), outputDataLength(frame).intValue());
-      frame.incrementRemainingGas(gasAvailableForChildCall(frame));
-      frame.popStackItems(getStackItemsConsumed());
-      frame.pushStackItem(Bytes32.ZERO);
-      return;
+  public OperationResult execute(final MessageFrame frame, final EVM evm) {
+    try {
+      // manual check becasue some reads won't come until the "complete" step.
+      if (frame.stackSize() < getStackItemsConsumed()) {
+        return UNDERFLOW_RESPONSE;
+      }
+      final Optional<ExceptionalHaltReason> haltReason = exceptionalHaltCondition(frame, evm);
+      if (haltReason.isPresent()) {
+        return new OperationResult(Optional.empty(), haltReason);
+      }
```

Comment:
Can this ever be true?

Reply:
`CallOperation` can have the exceptionalHaltCondition return an IllegalStateChange when in a static context with value.  I think I can get rid of this by having `CallOperation` check that first and then call the super.